templates/default.vcl.j2

```diff
@@ -9,3 +9,27 @@ backend default {
   .host = "{{ varnish_default_backend_host }}";
   .port = "{{ varnish_default_backend_port }}";
 }
+
+{% if varnish_backend is defined %}
+{% for backend, value in varnish_backend.iteritems() %}
+backend {{ backend }} {
+  .host = "{{ value.host }}";
+  .port = "{{ value.port }}";
+}
+{% endfor %}
+{% endif %}
+
+sub vcl_recv {
+
+{% if varnish_backend is defined %}
+{% for vhost, value in varnish_vhost.iteritems() if varnish_vhost is defined %}
```

Comment:
This is a little strange in terms of the logic—the `if varnish_vhost is defined` seems like it would be better to include in the main if condition on the line above (e.g. `if varnish_backend is defined and varnish_vhost is defined`)